source/common/router/scoped_rds.cc

```diff
@@ -363,6 +444,35 @@ ScopedRdsConfigSubscription::detectUpdateConflictAndCleanupRemoved(
   return clean_removed_resources;
 }
 
+void ScopedRdsConfigSubscription::onDemandRdsUpdate(
+    uint64_t key_hash, Event::Dispatcher& thread_local_dispatcher,
+    Http::RouteConfigUpdatedCallback&& route_config_updated_cb,
+    std::weak_ptr<Envoy::Config::ConfigSubscriptionCommonBase> weak_subscription) {
+  factory_context_.dispatcher().post([this, &thread_local_dispatcher, key_hash,
+                                      route_config_updated_cb, weak_subscription]() {
+    // If the subscription has been destroyed, return immediately.
+    if (!weak_subscription.lock()) {
+      thread_local_dispatcher.post([route_config_updated_cb] { route_config_updated_cb(false); });
```

Comment:
This is missing test coverage, see https://storage.googleapis.com/envoy-pr/12640/coverage/source/common/router/scoped_rds.cc.gcov.html. Can you validate that all the code you are adding/modifying has full coverage? Thanks.

Reply:
The missing test coverage is caused by a bug in my code. I didn't test the NullConfigImpl in the correct way. I have fixed it.